### DISCUSSION TITLE
How to access document metadata within an initialization of a custom template

### DISCUSSION THREAD

**Discussion by edmondop** [brechtm/rinohtype#344] [General]
I have a custom document template during which I would like to load a stylesheet dynamically. However, since I am using the Sphinx frontend, I do not invoke the constructor directly and I cannot pass attribute value since in my Sphinx .

I think also the documentation on how to use custom Template with template configuration in sphinx is fairly confusing :( Could anyone help please?

**Comment by brechtm** [comment]
> I have a custom document template during which I would like to load a stylesheet dynamically. However, since I am using the Sphinx frontend, I do not invoke the constructor directly and I cannot pass attribute value since in my Sphinx .

I'm not sure I understand the question. Let me try to rephrase:
- you have a custom template, i.e. Python code (not a template configuration .rtt file)
- you want to use this template, but with different style sheets, depending on ... ?

I haven't tried this, but this might work:
1. Add an identifier class attribute to your _DocumentTemplate_ subclass
    ```python

    class MyDocumentTemplate(DocumentTemplate):
        identifier = 'my_document_template'
        ...
   ```
2. Create your main template configuration file _my_template_cfg.rtt_
    ```ini
    [TEMPLATE_CONFIGURATION]
    name=my_template_cfg
    ...
    ```
3. Create several template configuration files that inherit from the main one, and only override the stylesheet
    ```ini
    [TEMPLATE_CONFIGURATION]
    name=template a
    base=my_template_cfg
    stylesheet=stylesheet_a.rts
    ```
4. In _conf.py_, point _template_ in a _rinoh_documents_ entry to the path of one of these template configuration files 

Or maybe it is better to point _template_ to a _TemplateConfiguration_ instance to avoid having to create multiple template configuration files. Perhaps something akin to:
    ```python
    TemplateConfiguration('bla', base='my_template_cfg.rtt', stylesheet='stylesheet_a.rts")
    ```

Please note that I haven't tested these, so it is not unlikely that you run into bugs. In that case, please be so kind to file a bug report. 😄

> I think also the documentation on how to use custom Template with template configuration in sphinx is fairly confusing :(

Yes, the documentation is lacking at the moment, I'm afraid. Can you be more specific as to how this could be improved?
